tests/unit/data-access.test.ts

```diff
@@ -0,0 +1,151 @@
+import { DataAccess } from '../../src/utils/data-access';
+import { TrailerRegistration } from '../../src/domain/trailer-registration';
+
+let mockDAO: DataAccess;
+describe('DataAccess', () => {
+  beforeEach(() => {
+    mockDAO = new DataAccess();
+  });
+
+  afterEach(() => {
+    jest.resetAllMocks().restoreAllMocks();
+  });
+
+  describe('getByVinOrChassisWithMake', () => {
+    test('should return record for the provided vinOrChassisWithMake', async () => {
+      const data = [
+        {
+          vinOrChassisWithMake: 'ABC1321234566big truck',
+          vin: 'ABC1321234566',
+          make: 'big truck',
+          trn: 'AB123AD',
+          certificateExpiryDate: '2021-12-12',
+          certificateIssueDate: '2021-01-01',
+        },
+      ];
+
+      const daoStub = jest.fn().mockImplementation(() => Promise.resolve({ Count: 1, Items: data }));
+
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
```

Comment:
I think we will need to catch up and probably make the linting more permissive, the dvsa eslint seems to be having rules stricter that eslint where some of them do not make sense for our case

Reply:
I agree, this makes testing really difficult.

Reply:
to be checked with Shaun